### PR TITLE
week1 solution faisalqureshi

### PR DESCRIPTION
**Pull request by QureshiFaisal**

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Awesome work. I will address a few comments once I try it out.

File: exercises/week 1/solutions/faisalqureshi.sh [hunk 1]
```diff
@@ -0,0 +1,171 @@
+LBTCL 
+
+
+# Verifying  the Binary
+
+VERSION="25.0"
+DMG_FILE="bitcoin-${VERSION}-x86_64-apple-darwin.dmg"
+CHECKSUMS_FILE="SHA256SUMS"
+SIGNATURE_FILE="SHA256SUMS.asc"
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/faisalqureshi.sh:9]: So here you are asking for the bitcoind binaries to be downloaded in a specific directory (the current working directory). The script should handle the downloading part of it too. Then the remaining process seems okay.

File: exercises/week 1/solutions/faisalqureshi.sh [hunk 2]
```diff
@@ -0,0 +1,171 @@
+LBTCL
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/faisalqureshi.sh:1]: This should be a comment..

File: exercises/week 1/solutions/faisalqureshi.sh [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    echo "❌ Binary signature verification unsuccessful! Please check the integrity of your binary. 😞"
+fi
+
+#Navigating to the bitcoin core directory
+
+cd /Volumes/SANDISK_64/bitcoin
+
+#Creating a bitcoin.conf file
+
+nano bitcoin.conf
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/faisalqureshi.sh:26]: This won't work. You need to first create the `~/.bitcoin/bitcoin.conf`file and then push the texts into them. You can search "how to write to a file in bash" in google.

File: exercises/week 1/solutions/faisalqureshi.sh [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+if shasum -a 256 -c "${CHECKSUMS_FILE}" 2>/dev/null | grep -q "${DMG_FILE}: OK"; then
+    echo "✅ Binary signature verification successful! Happy verifying! 😃"
+else
+    echo "❌ Binary signature verification unsuccessful! Please check the integrity of your binary. 😞"
+fi
+
+#Navigating to the bitcoin core directory
+
+cd /Volumes/SANDISK_64/bitcoin
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/faisalqureshi.sh:22]: This seems specific to your machine. The linux usual datadir is `~/.bitcoin`.

File: exercises/week 1/solutions/faisalqureshi.sh [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+bitcoin-cli  loadwallet Miner
+
+# Generating an address for Miner wallet with the label “Mining Reward”
+
+bitcoin-cli -rpcwallet=Miner getnewaddress "Mining Reward"
+
+# Mining Blocks
+
+bitcoin-cli  generatetoaddress 100 bcrt1q8fddy22r5ah7js3ukxtf4lf760qnuhc2hy348h
```
Comment (rajarshimaitra) [inline exercises/week 1/solutions/faisalqureshi.sh:59]: You won't get a spendable balance at 100 blocks. Need one more block.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
The current script isn't running, needs some below-mentioned modifications.

**Comment by QureshiFaisal** [comment]
Hi, thanks for the input. It's my first ever PR request, a lot of things that i did were specific to my system, and i also did not share a lot of workarounds that i did in the terminal to make it work. 
Will keep in mind that the script needs to be executable on all sytems, and will use variables instead of dynamic values wherever possible.

Your feedback is much appreciated.

**Review by Foxtrot-Zulu — APPROVED** [review]

**Pull request closed without merging**